exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1134 @@
+		if err != nil {
+			return nil, err
+		}
+		orders[i] = order.Detail{
+			Amount:          response[i].Size,
+			Pair:            pair,
+			Price:           response[i].Price,
+			ExecutedAmount:  response[i].FillSize,
+			RemainingAmount: response[i].Size - response[i].FillSize,
+			Fee:             response[i].FeeCurrency,
+			Exchange:        ok.Name,
+			OrderID:         response[i].OrderID,
+			ClientOrderID:   response[i].ClientSupplierOrderID,
+			Type:            oType,
+			Side:            orderSide,
+			Status:          orderStatus,
+			AssetType:       req.AssetType,
+			Date:            response[i].CreationTime,
+			LastUpdated:     response[i].UpdateTime,
+		}
+	}
+	return orders, nil
+}
+
+// GetFeeByType returns an estimate of fee based on the type of transaction
+func (ok *Okx) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBuilder) (float64, error) {
+	if feeBuilder == nil {
+		return 0, fmt.Errorf("%T %w", feeBuilder, common.ErrNilPointer)
+	}
+	if !ok.AreCredentialsValid(ctx) && feeBuilder.FeeType == exchange.CryptocurrencyTradeFee {
+		feeBuilder.FeeType = exchange.OfflineTradeFee
+	}
+	return ok.GetFee(ctx, feeBuilder)
+}
+
+// ValidateCredentials validates current credentials used for wrapper
+func (ok *Okx) ValidateCredentials(ctx context.Context, assetType asset.Item) error {
+	_, err := ok.UpdateAccountInfo(ctx, assetType)
+	return ok.CheckTransientError(err)
+}
+
+// GetHistoricCandles returns candles between a time period for a set time interval
+func (ok *Okx) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if err := ok.ValidateKline(pair, a, interval); err != nil {
+		return kline.Item{}, err
+	}
+	if kline.TotalCandlesPerInterval(start, end, interval) > float64(ok.Features.Enabled.Kline.ResultLimit) {
+		return kline.Item{}, errors.New(kline.ErrRequestExceedsExchangeLimits)
+	}
+	instrumentID, err := ok.getInstrumentIDFromPair(ctx, pair, a)
+	if err != nil {
+		return kline.Item{}, err
+	}
+	candles, err := ok.GetCandlesticksHistory(ctx, instrumentID, interval, time.Time{}, time.Time{}, 0)
+	if err != nil {
+		return kline.Item{}, err
+	}
+	response := kline.Item{
+		Exchange: ok.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+	for x := range candles {
+		response.Candles = append(response.Candles, kline.Candle{
+			Time:   candles[x].OpenTime,
+			Open:   candles[x].OpenPrice,
+			High:   candles[x].HighestPrice,
+			Low:    candles[x].LowestPrice,
+			Close:  candles[x].ClosePrice,
+			Volume: candles[x].Volume,
+		})
+	}
+	response.SortCandlesByTimestamp(false)
+	return response, nil
+}
+
+// GetHistoricCandlesExtended returns candles between a time period for a set time interval
+func (ok *Okx) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
```

Comment:
The documentation here definitely seems to suggest that this can be supported and implemented: https://www.okx.com/docs/en/#rest-api-market-data-get-candlesticks
The purpose of `GetHistoricCandlesExtended` is to ensure that all data between the start and end dates are requested, even through repeat requests and pagination